library/core/src/ffi/c_str.rs

```diff
@@ -322,8 +320,9 @@ impl CStr {
     /// assert_eq!(c_str.to_str().unwrap(), "AAAAAAAA");
     /// ```
     ///
-    #[unstable(feature = "cstr_from_bytes_until_nul", issue = "95027")]
-    #[rustc_const_unstable(feature = "cstr_from_bytes_until_nul", issue = "95027")]
+    #[rustc_allow_const_fn_unstable(const_slice_index)]
```

Comment:
As noticed in the follow up PR, this feature relies transitively on `const_trait_impls`, we should not stabilize anything based on them as it's not guaranteed we'll get that feature or an equivalent.

Reply:
Resolved in https://github.com/rust-lang/rust/pull/107624#discussion_r1101954066